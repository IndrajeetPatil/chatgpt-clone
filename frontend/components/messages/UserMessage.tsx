import React from "react";

import PersonIcon from "@mui/icons-material/Person";
import { Box, Paper, Typography, useTheme } from "@mui/material";

interface UserMessageProps {
  content: string;
}

const UserMessage: React.FC<UserMessageProps> = ({ content }) => {
  const theme = useTheme();
  const isDark = theme.palette.mode === "dark";

  return (
    <Box sx={{ display: "flex", justifyContent: "flex-end", mb: 2 }}>
      <Paper
        elevation={2}
        sx={{
          p: 2,
          backgroundColor: isDark ? "#1a237e" : "#e3f2fd",
          maxWidth: "70%",
          wordWrap: "break-word",
        }}
      >
        <PersonIcon sx={{ color: isDark ? "#90caf9" : "#1976d2" }} />
        <Typography
          variant="body1"
          component="div"
          sx={{ color: isDark ? "common.white" : "inherit" }}
        >
          {content}
        </Typography>
      </Paper>
    </Box>
  );
};

export default UserMessage;
