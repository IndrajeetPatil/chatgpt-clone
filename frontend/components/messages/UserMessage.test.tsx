import React from "react";
import { render } from "@testing-library/react";
import UserMessage from "./UserMessage";

// Mock Material-UI components to simplify snapshot
jest.mock("@mui/material", () => ({
  Box: ({
    children,
    ...props
  }: React.PropsWithChildren<Record<string, unknown>>) => (
    <div data-testid="mock-box" {...props}>
      {children}
    </div>
  ),
  Paper: ({
    children,
    ...props
  }: React.PropsWithChildren<Record<string, unknown>>) => (
    <div data-testid="mock-paper" {...props}>
      {children}
    </div>
  ),
  Typography: ({
    children,
    ...props
  }: React.PropsWithChildren<Record<string, unknown>>) => (
    <div data-testid="mock-typography" {...props}>
      {children}
    </div>
  ),
  useTheme: () => ({
    palette: {
      mode: "light",
    },
  }),
}));

// Mock PersonIcon
jest.mock("@mui/icons-material/Person", () => () => (
  <span data-testid="mock-person-icon">PersonIcon</span>
));

describe("UserMessage", () => {
  it("renders user message with person icon correctly", () => {
    const testContent = "Hello, world!";
    const { asFragment } = render(<UserMessage content={testContent} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
