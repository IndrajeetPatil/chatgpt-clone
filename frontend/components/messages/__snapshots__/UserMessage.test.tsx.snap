// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserMessage renders user message with person icon correctly 1`] = `
<DocumentFragment>
  <div
    data-testid="mock-box"
    sx="[object Object]"
  >
    <div
      data-testid="mock-paper"
      elevation="2"
      sx="[object Object]"
    >
      <span
        data-testid="mock-person-icon"
      >
        PersonIcon
      </span>
      <div
        component="div"
        data-testid="mock-typography"
        sx="[object Object]"
        variant="body1"
      >
        Hello, world!
      </div>
    </div>
  </div>
</DocumentFragment>
`;
